clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -2608,8 +2608,15 @@ void CIRGenModule::setFunctionAttributes(GlobalDecl globalDecl,
 
   // TODO(cir): Complete the remaining part of the function.
   assert(!MissingFeatures::setFunctionAttributes());
-  auto decl = globalDecl.getDecl();
-  func.setGlobalVisibilityAttr(getGlobalVisibilityAttrFromDecl(decl));
+
+  // TODO(cir): This needs a lot of work to better match CodeGen. That
+  // ultimately ends up in setGlobalVisibility, which already has the linkage of
+  // the LLVM GV (corresponding to our FuncOp) computed, so it doesn't have to
+  // recompute it here. This is a minimal fix for now.
```

Comment:
Looks like this is subtle enough that it deserves an issue to track it?

Reply:
Yup, I filed #992 to fix this up more thoroughly.